compiler/rustc_mir/src/transform/check_consts/validation.rs

```diff
@@ -508,20 +508,20 @@ impl Visitor<'tcx> for Validator<'mir, 'tcx> {
                         }
                     };
                     self.visit_local(&place.local, ctx, location);
-                    self.visit_projection(place.local, reborrowed_proj, ctx, location);
+                    self.visit_projection(place.local, place_ref.projection, ctx, location);
```

Comment:
Same here, `place_ref.local`.

Really `Visitor` should be changed such that `` takes a `PlaceRef` instead of taking the local and the projections separately.